emission/net/int_service/giles/archiver.py

```diff
@@ -34,16 +36,20 @@ def __init__(self, collection):
     # NOTE: When we insert an entry to the Archiver, we associate that entry with a stream UUID, which is
     # a function of the stat and user_uuid. 
     # No mapping needs to be maintained since this UUID can be reconstructed as long as the above data is found.
+    @timeout.timeout(seconds=60)
     def insert(self, entry):
         assert type(entry) == dict
-        stat = entry['stat']
-        user_uuid = str(entry['user'])
-
-        # TODO: Support more precise timestamps
-        # Giles has some problem unmarshalling floats
-        #client_ts = int(entry['ts'])
-        client_ts = int(entry['ts'])
-        reading = entry['reading']
+        
+        try:
+            stat = str(entry['stat'])
+            user_uuid = str(entry['user'])
+            # TODO: Support more precise timestamps
+            # Giles has some problem unmarshalling floats
+            #client_ts = int(entry['ts'])
+            client_ts = int(entry['ts'])
+            reading = float(entry['reading'])
+        except Exception as e:
```

Comment:
That's true. I added a try/catch in the archiver so that I could write the negative test case. That way the archiver itself has ways to deal with failure independently of how it's being called.